test/unit/middlewares/egress-tracker.spec.js

```diff
@@ -0,0 +1,462 @@
+/* eslint-disable no-unused-expressions
+   ---
+   `no-unused-expressions` doesn't understand that several of Chai's assertions
+   are implemented as getters rather than explicit function calls; it thinks
+   the assertions are unused expressions. */
+import { randomBytes } from 'node:crypto'
+import { describe, it, afterEach, before } from 'mocha'
+import { assert, expect } from 'chai'
+import sinon from 'sinon'
+import { CID } from 'multiformats'
+import { withEgressHandler } from '../../../src/handlers/egress-tracker.js'
+import { Builder, toBlobKey } from '../../helpers/builder.js'
+import { CARReaderStream } from 'carstream'
+
+/**
+ * Creates a request with an optional authorization header.
+ *
+ * @param {Object} [options]
+ * @param {string} [options.authorization] The value for the `Authorization`
+ * header, if any.
+ */
+const createRequest = async ({ authorization } = {}) =>
+  new Request('http://doesnt-matter.com/', {
+    headers: new Headers(
+      authorization ? { Authorization: authorization } : {}
+    )
+  })
+
+const env =
+  /** @satisfies {import('../../../src/handlers/egress-tracker.types.js').Environment} */
+  ({
+    DEBUG: 'true',
+    ACCOUNTING_SERVICE_URL: 'http://example.com',
+    FF_EGRESS_TRACKER_ENABLED: 'true'
+  })
+
+const accountingRecordMethodStub = sinon.stub()
+  // @ts-expect-error
+  .returns(async (cid, bytes, servedAt) => {
+    console.log(`[mock] record called with cid: ${cid}, bytes: ${bytes}, servedAt: ${servedAt}`)
+    return Promise.resolve()
```

Comment:
**suggestion:** If I'm reading right, we can just drop this line altogether. It's the same as just `return`.

Reply:
Async funcs automatically convert return values to promises, so yes.